core/src/main/java/org/apache/calcite/jdbc/SimpleCalciteSchema.java

```diff
@@ -67,31 +68,64 @@ public CalciteSchema add(String name, Schema schema) {
     return calciteSchema;
   }
 
+  private String caseInsensitiveLookup(Set<String> candidates, String name) {
+    for (String candidate: candidates) {
```

Comment:
I suggest it should try:
1) exact lookup
2) name.toUpperCase() lookup
3) name.toLowerCase() lookup

otherwise => iterate over the set.

@DonnyZone , WDYT?

Reply:
I agree. It makes sense for most frequent cases, i.e., upper case or lower case.
I will add this optimization soon. Thanks!